### PR TITLE
add chunking to grpc to avoid large response is rejected by the client.

### PR DESCRIPTION
**Pull request by larry-aptos**
### Description
* add chunking to grpc to avoid large response is rejected by the client.

### CODE REVIEW

**Review by bowenyang007 — APPROVED** [review]

File: ecosystem/indexer-grpc/indexer-grpc-utils/src/lib.rs [hunk 1]
```diff
@@ -60,3 +62,73 @@ pub fn time_diff_since_pb_timestamp_in_secs(timestamp: &Timestamp) -> f64 {
...
+            }),
+            ..Transaction::default()
+        };
+        // Create a vec with 10 transactions.
+        let transactions = vec![t.clone(); 10];
+        assert!(t.encoded_len() > 1);
+        let chunked_transactions = chunk_transactions(transactions, 1);
+        assert_eq!(chunked_transactions.len(), 10);
+    }
+    #[test]
```
Comment (bowenyang007) [inline ecosystem/indexer-grpc/indexer-grpc-utils/src/lib.rs:113]: extra line

File: ecosystem/indexer-grpc/indexer-grpc-fullnode/src/stream_coordinator.rs [hunk 2]
```diff
@@ -24,6 +25,8 @@ use tokio::sync::mpsc;
 use tonic::Status;
 
 type EndVersion = u64;
+// Limit the message size to 15MB. By default the downstream can receive up to 15MB.
+const MESSAGE_SIZE_LIMIT: usize = 1024 * 1024 * 15;
```
Comment (bowenyang007) [inline ecosystem/indexer-grpc/indexer-grpc-fullnode/src/stream_coordinator.rs:29]: let's use only 1 constant.

**Review by bowenyang007 — APPROVED** [review]

**Comment by bowenyang007** [comment]
Btw we should add in docs somewhere that the max size is 80MB. Downstream I would want to keep a buffer and let's say I want 100 batches. That would be 8GB of memory and maybe not practical, so this is an important piece of data.

File: ecosystem/indexer-grpc/indexer-grpc-utils/src/lib.rs (hunk 1 above)
Comment (larry-aptos) [reply ecosystem/indexer-grpc/indexer-grpc-utils/src/lib.rs:114]: done.

File: ecosystem/indexer-grpc/indexer-grpc-fullnode/src/stream_coordinator.rs (hunk 2 above)
Comment (larry-aptos) [reply ecosystem/indexer-grpc/indexer-grpc-fullnode/src/stream_coordinator.rs]: done.

**Pull request merged**